pkg/gitops/rollback_test.go

```diff
@@ -0,0 +1,224 @@
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+			Egress:      []networkingv1.NetworkPolicyEgressRule{{}},
+		},
+	}
+	anp1 = &crdv1alpha1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "crd.antrea.io/v1alpha1"},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "anpA", UID: "uidC"},
+		Spec: crdv1alpha1.NetworkPolicySpec{
+			AppliedTo: []crdv1alpha1.NetworkPolicyPeer{
+				{PodSelector: &selectorA},
+			},
+			Priority: p10,
+			Ingress: []crdv1alpha1.Rule{
+				{
+					Ports: []crdv1alpha1.NetworkPolicyPort{
+						{
+							Port: &int80,
+						},
+					},
+					From: []crdv1alpha1.NetworkPolicyPeer{
+						{
+							PodSelector:       &selectorB,
+							NamespaceSelector: &selectorC,
+						},
+					},
+					Action: &allowAction,
+				},
+			},
+			Egress: []crdv1alpha1.Rule{
+				{
+					Ports: []crdv1alpha1.NetworkPolicyPort{
+						{
+							Port: &int81,
+						},
+					},
+					To: []crdv1alpha1.NetworkPolicyPeer{
+						{
+							PodSelector:       &selectorB,
+							NamespaceSelector: &selectorC,
+						},
+					},
+					Action: &allowAction,
+				},
+			},
+		},
+	}
+)
+
+func TestRollback(t *testing.T) {
+	fakeClient := NewClient(np1.DeepCopy(), anp1.DeepCopy())
+	k8s := &K8sClient{
+		Client: fakeClient,
+	}
+	cr, err := SetupRepo(k8s, StorageModeInMemory, dir)
+	if err != nil {
+		t.Errorf("Error (TestRollback): unable to set up repo")
+	}
+	h, err := cr.Repo.Head()
+	if err != nil {
+		t.Errorf("Error (TestRollback): unable to get repo head ref")
+	}
+	testSig := &object.Signature{
+		Name:  "test",
+		Email: "test@antrea.audit.io",
+		When:  time.Now(),
+	}
+	if err := cr.TagCommit(h.Hash().String(), "test-tag", testSig); err != nil {
+		t.Errorf("Error (TestRollback): unable to create new tag")
+	}
+
+	// Create, update, and delete a resource
+	r := unstructured.Unstructured{}
+	r.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "networking.k8s.io",
+		Version: "v1",
+		Kind:    "NetworkPolicy",
+	})
+	j, err := json.Marshal(np2)
+	if err != nil {
+		t.Errorf("Error (TestRollback): unable to convert to json")
+	}
+	if err := json.Unmarshal(j, &r); err != nil {
+		t.Errorf("Error (TestRollback): unable to unmarshal into unstructured object")
+	}
+	if err := k8s.CreateOrUpdateResource(&r); err != nil {
+		t.Errorf("Error (TestRollback): unable to create new resource")
+	}
+
+	updatedNP := np1
+	updatedNP.ObjectMeta.SetClusterName("new-cluster-name")
+	r = unstructured.Unstructured{}
+	r.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "networking.k8s.io",
+		Version: "v1",
+		Kind:    "NetworkPolicy",
+	})
+	j, err = json.Marshal(updatedNP)
+	if err != nil {
+		t.Errorf("Error (TestRollback): unable to convert to json")
+	}
+	if err := json.Unmarshal(j, &r); err != nil {
+		t.Errorf("Error (TestRollback): unable to unmarshal into unstructured object")
+	}
```

Comment:
is there a less verbose way to do these conversions? How about `runtime.DefaultUnstructuredConverter.ToUnstructured(item)`?